pkg/ui/workspaces/cluster-ui/src/api/fetchData.ts

```diff
@@ -60,38 +63,22 @@ export const fetchData = <P extends ProtoBuilder<P>, T extends ProtoBuilder<T>>(
     params.body = toArrayBuffer(encodedRequest);
   }
 
-  return fetch(withBasePath(path), params)
-    .then(response => {
-      if (!response.ok) {
-        return response.arrayBuffer().then(buffer => {
-          let respError;
-          try {
-            respError = cockroach.server.serverpb.ResponseError.decode(
-              new Uint8Array(buffer),
-            );
-          } catch {
-            respError = new cockroach.server.serverpb.ResponseError({
-              error: response.statusText,
-            });
-          }
-          throw new RequestError(
-            response.statusText,
-            response.status,
-            respError.error,
-          );
-        });
-      }
-      return response.arrayBuffer();
-    })
-    .then(buffer => respBuilder.decode(new Uint8Array(buffer)));
+  const response = await fetch(withBasePath(path), params);
+  if (!response.ok) {
+    const errorBody = await getErrorBodyFromResponse(response);
+    throw new RequestError(response.status, errorBody);
+  }
+
+  const buffer = await response.arrayBuffer();
+  return respBuilder.decode(new Uint8Array(buffer));
 };
 
 /**
  * fetchDataJSON makes a request for /api/v2 which uses content type JSON.
  * @param path relative path for requested resource.
  * @param reqPayload request payload object.
  */
-export function fetchDataJSON<ResponseType, RequestType>(
+export async function fetchDataJSON<ResponseType, RequestType>(
```

Comment:
why do we do `export async function` here, but `export const fetchData = async` above? Is there a difference? Both return promises.

Reply:
There's no difference for our use cases of these functions. For context `fetchDataJSON` was introduced by me at a   much later date and I have a personal preference to use the `function` syntax in the global scope 😅 
I don't believe we have a strong reason to enforce one over the other right now.